php/class-photonfill.php

```diff
@@ -369,9 +369,9 @@ public function create_image_object( $attachment_id, $current_size, $args = arra
 		public function add_image_metadata( $data, $attachment_id ) {
 			foreach ( $this->image_sizes as $size => $breakpoint ) {
 				$data['sizes'][ $size ] = array(
-					'file' => wp_basename( $data['file'] ),
-					'width' => $data['width'],
-					'height' => $data['height'],
+					'file' => ( ! empty( $data['file'] ) ) ? wp_basename( $data['file'] ) : '',
+					'width' => ( ! empty( $data['width'] ) ) ? $data['width'] : '',
+					'height' => ( ! empty( $data['height'] ) ) ? $data['height'] : '',
```

Comment:
not a huge deal but it might make sense to just dump these into parameters before the loop and use those, since all the checks are based on data from outside the loop